stingray/lightcurve.py

```diff
@@ -151,74 +183,146 @@ def __init__(self, time, counts, err=None, input_counts=True,
                 raise StingrayError("Statistic not recognized."
                                     "Please select one of these: ",
                                     "{}".format(valid_statistics))
-            if err_dist.lower() == 'poisson':
-                # Instead of the simple square root, we use confidence
-                # intervals (should be valid for low fluxes too)
+
+            if precise_poisson_err:
                 err = poisson_symmetrical_errors(counts)
-            else:
+
+            if not err_dist.lower() == 'poisson':
                 simon("Stingray only uses poisson err_dist at the moment, "
                       "We are setting your errors to zero. "
                       "Sorry for the inconvenience.")
                 err = np.zeros_like(counts)
```

Comment:
I am very confused by the logic here, because there are a number of consecutive if statements, at least some of which should be if-else statements, I think, because they're mutually exclusive. 

So, what's the difference between `if err_dist.lower() not in valid_statistics:` and `if not err_dist.lower() == 'poisson':`. Shouldn't our `valid_statistics` currently be only Poisson if that's all we've got implemented? Why are we asking this twice?

Secondly, it seems to me that `if precise_poisson_err` ought to be nested within that larger if-else statement (is it Poisson or not)? Under the current code, what happens if `err_dist.lower() == 'poisson'`, but `precise_poisson_err == False`?

Reply:
Oh, right.
I think that `valid_statistics` should include at least poisson and gauss, without setting the errors to zero in the gauss thing. I suspect this was to avoid some undocumented side effect. I would not set the errors to zero, but give the warning. Basically, "dear friend, you can specify the error bars here, but it will only be useful to plot your light curve, any other operation will ignore them."
`precise_poisson_error` is more precise than "sqrt(mean(counts))" but slower. However, not too much slower. So I think this option can be safely eliminated.